### PR TITLE
Reset Kit Cooldowns

### PR DESCRIPTION
**Pull request by darbyjack**
This is not a finished PR. I'm just opening it so we can keep track of progress.

### CODE REVIEW

**Comment by darbyjack** [comment]
Update: Have given this out for people to test and they've given good feedback.

File: Essentials/src/plugin.yml [hunk 1]
```diff
@@ -240,6 +240,10 @@ commands:
     description: Throw an exploding kitten at your opponent.
     usage: /<command>
     aliases: [ekittycannon]
+  kitreset:
+    description: Reset the cooldown of a kit for a player.
+    usage: /<command> [player] [kit]
+    aliases: [kr, ekr]
```
Comment (mdcfe) [inline Essentials/src/plugin.yml]: Needs to include `/ekitreset`.

Comment (darbyjack) [reply Essentials/src/plugin.yml]: Done!

**Review by Ichbinjoe — APPROVED** [review]

File: Essentials/src/com/earth2me/essentials/commands/Commandkitreset.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        sender.sendMessage(tl("kitReset", userTo.getDisplayName(), kit.getName()));
+        userTo.sendMessage(tl("kitResetPlayer", kit.getName()));
+    }
+
+    @Override
+    protected List<String> getTabCompleteOptions(final Server server, final User user, final String commandLabel, final String[] args) {
+        List<String> options = new ArrayList<>();
+        if (args.length == 1 && user.isAuthorized("essentials.kit.reset")) {
+            return getPlayers(server, user);
+        } else if (args.length == 2)  {
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/commands/Commandkitreset.java]: Missing second essentials.kit.reset permission check?

File: Essentials/src/com/earth2me/essentials/commands/Commandkitreset.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            for (String kitName : ess.getKits().getKits().getKeys(false)) {
+                if (!user.isAuthorized("essentials.kits." + kitName)) { // Only check perm, not time or money
+                    continue;
+                }
+                options.add(kitName);
+            }
+            return options;
+        }
+        return Collections.emptyList();
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/commands/Commandkitreset.java]: return options;, you already constructed it, why not use it!

**Review by Ichbinjoe — CHANGES_REQUESTED** [review]

File: Essentials/src/com/earth2me/essentials/commands/Commandkitreset.java (hunk 3 above)
Comment (darbyjack) [reply Essentials/src/com/earth2me/essentials/commands/Commandkitreset.java]: I think I rebased this correctly. Added the changes.

**Comment by Ichbinjoe** [comment]
lmao this pr is fucked

**Pull request closed without merging**